packages/alea-frontend/pages/api/course-metadata/get-homework-and-quiz.ts

```diff
@@ -0,0 +1,25 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { executeAndEndSet500OnError } from '../comment-utils';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const { courseId, instanceId } = req.query;
+
+  if (!courseId) {
+    res.status(422).end('Missing required fields');
+    return;
+  }
+
+  const result = await executeAndEndSet500OnError(
+    `SELECT hasQuiz, hasHomework FROM courseMetadata WHERE courseId = ? AND instanceId = ?`,
+    [courseId, instanceId],
+    res
+  );
+
+  if (!result?.length) {
+    return res.status(200).json({ hasQuiz: false, hasHomework: false });
+  }
```

Comment:
Must do falsy early return after an error-handler

https://github.com/KWARC/ALeA/blob/main/backend_and_general_guidelines.md#27-returning-api-errors-in-helper-functions-not-directly-in-the-handler-function


```suggestion
  if (!result) return; // This is for database error, status 500 was already sent by the function if 'result' is falsy
  if (!result.length) { // this is the case when no db error but the db didn't contain any entry 
    return res.status(200).json({ hasQuiz: false, hasHomework: false });
  }
  
```